Include "kubevirt-web-ui" playbook in the "kubevirt"

mareklibra opened the pull request:
**What**: Include `kubevirt-web-ui` playbook from the `kubevirt`

The `kubevirt/web-ui` is mandatory part of kubevirt deployment on the OpneShift.

**Special notes for your reviewer**:
Requires: #417 
This needs to be verified in different setups.

**Release note**:
```release-note
The Kubevirt Web UI is installed by default along Kubevirt on OpenShift.
```

mareklibra commented:
@rthallisey , is this the right place for kubevirt-web-ui inclusion within OpenShift deployment?

mareklibra commented:
@rthallisey , is this the right place for kubevirt-web-ui inclusion within OpenShift deployment?

mareklibra commented:
@rthallisey , is this the right place for kubevirt-web-ui inclusion within OpenShift deployment?

mareklibra commented:
@rthallisey , is this the right place for kubevirt-web-ui inclusion within OpenShift deployment?

mareklibra commented:
@rthallisey , is this the right place for kubevirt-web-ui inclusion within OpenShift deployment?

mareklibra commented:
@rthallisey , is this the right place for kubevirt-web-ui inclusion within OpenShift deployment?

rthallisey commented on inventory:
```diff
@@ -16,6 +16,11 @@ openshift_install_examples=false
 openshift_master_default_subdomain=cloudapps.example.com
 
+# Image used for the kubevirt/web-ui deployment
+kubevirt_web_ui_image_name=docker.io/mareklibra/kubevirt-web-ui:v1.3-8
```
talk to @fabiand about getting this published to docker.io/kubevirt

mareklibra replied:
Sure. This will take some time, can we fix that in a follow-up, please?

rthallisey commented on playbooks/kubevirt-web-ui/private/config.yml:
```diff
@@ -12,18 +12,19 @@
           playbook: "playbooks/kubevirt-web-ui/config.yml"
           status: "In Progress"
           start: "{{ lookup('pipe', 'date +%Y%m%d%H%M%SZ') }}"
+    when: cluster | default("openshift") =="openshift"
```
you shouldn't have to worry about cluster being set.  If you're seeing not getting set, then cluster should be defaulted to openshift elsewhere.

mareklibra replied:
ok, removed from here, the variable must be set to install Kubevirt Web UI.
I've added a debug message on the top of this file if `cluster` is missing.

mareklibra replied:
CI is failing on undefined `cluster` during uninstallation.

@rthallisey , are you fine with adding this default back? (see 2nd commit)
To fix deprovision phase farther, I've cherry-picked #423 here.

rthallisey commented on roles/kubevirt_web_ui/tasks/main.yml:
```diff
@@ -2,7 +2,7 @@
 
 - include_tasks: install.yml
-  when: kubevirt_web_ui_install | default(true) | bool
+  when: apb_action | default("provision") | string == "provision"
```
if you rename install.yaml and remove.yml to 'provision' and 'deprovision' you can do:
```yaml
- include_tasks: {{ apb_action}}.yml
```

gbenhaim commented:
@mareklibra 
```
19:46:17 [check-patch.openshift_3-10.el7.x86_64] TASK [kubevirt_web_ui : include_tasks] *****************************************
19:46:17 [check-patch.openshift_3-10.el7.x86_64] Sunday 28 October 2018  19:46:17 +0000 (0:00:01.428)       0:39:49.956 ******** 
19:46:17 [check-patch.openshift_3-10.el7.x86_64] fatal: [lago-master]: FAILED! => {}
19:46:17 [check-patch.openshift_3-10.el7.x86_64] 
19:46:17 [check-patch.openshift_3-10.el7.x86_64] MSG:
19:46:17 [check-patch.openshift_3-10.el7.x86_64] 
19:46:17 [check-patch.openshift_3-10.el7.x86_64] 'apb_action' is undefined
```
can you please check?

mareklibra commented:
Thanks, rebased and added `apb_action: "provision"` to defaults.yaml .

Recently failing CI is most probably caused by failing scheduler as the pod is expected to run on a master node (as openshift-console). Thanks to auto-addition of `defaultNodeSelector` value, there's no "master and compute" node.
Trying to schedule the pod on a "compute" node only.

mareklibra commented:
Seems to be working in my local test environment, so pushing to try on CI

mareklibra commented:
The issue with node selector was fixed.

CI is recently failing on deprovision:
- the `cluster` is undefined in playbook, so adding back defaults
- #423 fixes are required, so cherry-picking here

@rthallisey , wdyt?

rthallisey commented:
That's fine.  `cluster` should really be defaulted in vars/all.yml, it wouldn't hurt to have them defaulted elsewhere for now.

gbenhaim commented:
@rthallisey Can we merge it?
